pydruid/query.py

```diff
@@ -266,20 +267,52 @@ def build_query(self, query_type, args):
             elif key == 'datasource':
                 query_dict['dataSource'] = self.parse_datasource(val, query_type)
             elif key == 'paging_spec':
+                # TODO: should paging only exist in outer query?
                 query_dict['pagingSpec'] = val
             elif key == 'limit_spec':
+                # TODO: should limiting only exist in outer query?
                 query_dict['limitSpec'] = val
             elif key == "filter" and val is not None:
                 query_dict[key] = Filter.build_filter(val)
             elif key == "having" and val is not None:
                 query_dict[key] = Having.build_having(val)
             elif key == 'dimension' and val is not None:
-                query_dict[key] = build_dimension(val)
+                if 'inner_dimensions' in args:
+                    query_dict['dimensions'] = [build_dimension(v) for v in args['inner_dimensions']]
+                else:
+                    query_dict[key] = build_dimension(val)
             elif key == 'dimensions':
-                query_dict[key] = [build_dimension(v) for v in val]
+                if 'inner_dimensions' in args:
+                    query_dict[key] = [build_dimension(v) for v in args['inner_dimensions']]
+                else:
+                    query_dict[key] = [build_dimension(v) for v in val]
+            elif key == 'inner_dimensions':
+                continue
             else:
                 query_dict[key] = val
 
+        if 'inner_dimensions' in args:
```

Comment:
Can you give some more details on why we chose to implement the "inner_dimensions" arg and duplicate mot of build_query code? 

I was thinking that maybe we can get datasource of type "inner_query" (or something like that 😅) and use the build_query function to build also the inner query for the datasource. WDYT? 😄